cmd/build/helmify/main.go

```diff
@@ -142,13 +142,17 @@ func (ks *kindSet) Write() error {
 			}
 
 			if name == "gatekeeper-controller-manager" && kind == "PodDisruptionBudget" {
-				obj = strings.Replace(obj, "apiVersion: policy/v1beta1", "{{- if .Capabilities.APIVersions.Has \"policy/v1\" }}\napiVersion: policy/v1\n{{ else }}\napiVersion: policy/v1beta1\n{{ end -}}", 1)
+				obj = strings.Replace(obj, "apiVersion: policy/v1beta1", "{{- $v1 := .Capabilities.APIVersions.Has \"policy/v1/PodDisruptionBudget\" -}}\n{{- $v1beta1 := .Capabilities.APIVersions.Has \"policy/v1beta1/PodDisruptionBudget\" -}}\napiVersion: policy/v1{{- if and (not $v1) $v1beta1 -}}beta1{{- end }}", 1)
 			}
 
 			if name == "gatekeeper-admin" && kind == "PodSecurityPolicy" {
 				obj = "{{- if .Values.psp.enabled }}\n" + obj + "{{- end }}\n"
 			}
 
+			if name == "gatekeeper-manager-role" && kind == "ClusterRole" {
+				obj = strings.Replace(obj, "- apiGroups:\n  - policy\n  resourceNames:\n  - gatekeeper-admin\n  resources:\n  - podsecuritypolicies\n  verbs:\n  - use\n", "{{- if .Values.psp.enabled }}\n- apiGroups:\n  - policy\n  resourceNames:\n  - gatekeeper-admin\n  resources:\n  - podsecuritypolicies\n  verbs:\n  - use\n{{- end }}\n", 1)
```

Comment:
can we add a policy/v1beta1 version check here too?

Reply:
@sozercan I'm not sure I understand what the version check will be for.  Lines 148-150 will remove the PSP if `psp.enabled: false` is set, regardless of the API version.  These lines will remove the use of the PSP regardless of API.

Maybe this is getting confused with the change in lines 144-146, which refer to the PodDistruptionBudget rather than the PodSecurityPolicy?

Reply:
since PSP will be removed in K8s 1.25, we want to make sure to skip the role if `policy/v1beta1/PodSecurityPolicy` doesn't exist even if PSP is enabled

Reply:
I can also add this to be part of #2174 if we want to defer this piece

Reply:
Resolved with https://github.com/open-policy-agent/gatekeeper/pull/2187/commits/ec1d5e722b2f04068b14203b9eac7b43c7666d04